x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/execution/sample/CircuitBreakerTests.java

```diff
@@ -0,0 +1,335 @@
+            List.of("foo", "bar"),
+            List.of(new FieldAttribute(Source.EMPTY, "foo", new EsField("foo", DataTypes.KEYWORD, null, false))),
+            randomIntBetween(10, 500)
+        );
+    }
+
+    private List<SampleCriterion> mockCriteria() {
+        return List.of(new SampleCriterion(mockQueryRequest(), mockQueryRequest(), mockQueryRequest(), emptyList(), emptyList()));
+    }
+
+    private class ESMockClient extends NoOpClient {
+        protected final CircuitBreaker circuitBreaker;
+        private final String pitId = "test_pit_id";
+
+        ESMockClient(CircuitBreaker circuitBreaker) {
+            super(getTestName());
+            this.circuitBreaker = circuitBreaker;
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        protected <Request extends ActionRequest, Response extends ActionResponse> void doExecute(
+            ActionType<Response> action,
+            Request request,
+            ActionListener<Response> listener
+        ) {
+            if (request instanceof OpenPointInTimeRequest) {
+                OpenPointInTimeResponse response = new OpenPointInTimeResponse(pitId);
+                listener.onResponse((Response) response);
+            } else if (request instanceof ClosePointInTimeRequest) {
+                ClosePointInTimeResponse response = new ClosePointInTimeResponse(true, 1);
+                listener.onResponse((Response) response);
+            } else if (request instanceof SearchRequest searchRequest) {
+                handleSearchRequest(listener, searchRequest);
+            } else {
+                super.doExecute(action, request, listener);
+            }
+        }
+
+        @SuppressWarnings("unchecked")
+        <Response extends ActionResponse> void handleSearchRequest(ActionListener<Response> listener, SearchRequest searchRequest) {
+            Aggregations aggs = new Aggregations(List.of(newInternalComposite()));
+
+            SearchResponseSections internal = new SearchResponseSections(null, aggs, null, false, false, null, 0);
+            SearchResponse response = new SearchResponse(
+                internal,
+                null,
+                2,
+                0,
+                0,
+                0,
+                ShardSearchFailure.EMPTY_ARRAY,
+                Clusters.EMPTY,
+                searchRequest.pointInTimeBuilder().getEncodedId()
+            );
+
+            listener.onResponse((Response) response);
+        }
+
+        private InternalComposite newInternalComposite() {
+            // InternalComposite(name, size, ...) is not public
+            try {
+                return new InternalComposite(new StreamInput() {
+                    @Override
+                    public byte readByte() throws IOException {
+                        return 0;
+                    }
+
+                    @Override
+                    public void readBytes(byte[] b, int offset, int len) throws IOException {
+
```

Comment:
Nit: could this empty methods be collapsed on one line?

Reply:
Unfortunately our style check rules complain about that